auth.go

```diff
@@ -572,3 +572,11 @@ func fillCachedIDToken(sc *snowflakeConn) {
 func fillCachedMfaToken(sc *snowflakeConn) {
 	getCredential(sc, mfaToken)
 }
+
+func getAccountForJWT(accountName string) string {
```

Comment:
This function looks more general, maybe we can:
1. rename it to `getAccountNameOnly` or something like this (this name is bad, so if you have better idea - feel free!)
2. move it to dsn.go
3. I'm considering having it as a member function of `Config`, but probably this is not for the best...
WDYT?

Reply:
I'd have to agree with @sfc-gh-pfus - signature suggests that the it's logic is strongly coupled with JWT concept, while it's not. My suggestion is `extractAccountName(rawAccount string)`.
Still, mad respect for @sfc-gh-dszmolka for standing up for himself!

Reply:
the goal is really to strongly couple with JWT generation, the fix is targeting this area as the issue stems from wrong JWT

renamed and moved the function per the suggestions.